BuildingBlocks/test/BuildingBlocks.Application.Tests/Mediatr/QuotaEnforcerBehaviorTests.cs

```diff
@@ -109,8 +109,9 @@ public void Throws_QuotaExhaustedException_when_the_metric_is_still_exhausted()
             CancellationToken.None);
 
         // Assert
-        acting.Should().AwaitThrowAsync<QuotaExhaustedException>()
-            .Which.ExhaustedMetricStatuses.First().MetricKey.Should().Be(TestData.MetricStatus.ThatIsExhaustedFor10Days.MetricKey);
+        var exceptionExhaustedMetrics = acting.Should().AwaitThrowAsync<QuotaExhaustedException>().Which.ExhaustedMetricStatuses;
+        exceptionExhaustedMetrics.First().MetricKey.Should().Be(TestData.MetricStatus.ThatIsExhaustedFor10Days.MetricKey);
+        exceptionExhaustedMetrics.Should().HaveCount(1);
```

Comment:
Change order so that we get an assertion error instead of an `InvalidOperationException` caused by `First()`. Same for the other occurrences.